src/components/range-inputs.tsx

```diff
@@ -0,0 +1,234 @@
+        if (!value) return true;
+
+        if (type === 'int') return INTEGER_REGEXP.test(value);
+
+        if (type === 'float') return FLOAT_REGEXP.test(value);
+
+        /**in case type is timestamp and both date n time values are null */
+        if (type === 'timestamp' && !value) return true;
+
+        /**type is either date or timestamp */
+        const formatString = type === 'date' ? DATE_FORMAT : TIMESTAMP_FORMAT;
+        const date = momentJS(value, formatString, true);
+        return date.isValid();
+    }
+
+    /**
+     * performs basic range validation : from_value > to_value
+    */
+    const rangeCheck = (fromVal: string, toVal: string): boolean => {
+        if (type === 'int') return parseInt(fromVal) < parseInt(toVal);
+
+        if (type === 'float') return parseInt(fromVal) < parseInt(toVal);
+
+        /**type is either date or timestamp */
+        const formatString = type === 'date' ? DATE_FORMAT : TIMESTAMP_FORMAT;
+        const fromDate = momentJS(fromVal, formatString, true);
+        const toDate = momentJS(toVal, formatString, true);
+        return toDate.diff(fromDate) > 0
+    }
+
+    const validateValues = (fromVal: string, toVal: string): string => {
+        const isfromValid = validateValue(fromVal);
+
+        const isToValid = validateValue(toVal);
+
+        if (!isfromValid || !isToValid) return type;
+
+        /**if only one field is non null then don't perform range check */
+        if (!fromVal || !toVal) return 'valid';
+
+        /**both from and to values are now valid so perform range validations */
+        return rangeCheck(fromVal, toVal) ? 'valid' : 'range';
+    }
+
+    const handleSubmit = () => {
+        const formatedValues = type === 'timestamp' ? formatTimeValues()
+            : { fromVal: fromRef?.current?.value || '', toVal: toRef?.current?.value || '' };
+
+        const validatedResult = validateValues(formatedValues.fromVal, formatedValues.toVal);
+
+        console.log('validation result: ', validatedResult);
+
+        if (validatedResult === 'valid') {
+            /* clear out any previous errors when a new submission is validated */
+            setError(null);
+
+            /* eslint-disable  @typescript-eslint/no-non-null-assertion */
+            console.log('values sent to server', formatedValues);
+            /* eslint-enable  @typescript-eslint/no-non-null-assertion */
+        } else {
+            setError(errorMsgMap[validatedResult]);
+        }
+    };
+
+    const classTypeName = (type === 'int' || type === 'float') ? 'numeric-width' : type === 'date' ? 'date-width' : 'time-width';
+
+    return (
+        <div className={classes}>
+            <div className='range-input-container'>
+                <div className={`range-input ${classTypeName}`}>
+                    <label htmlFor={`range-from-val-${type}`}>From:</label>
+                    <InputSwitch
+                        id={`range-from-val-${type}`}
```

Comment:
What's the purpose of this `id`? Did you add it only for `htmlFor` of the label?

One mistake that we made in AngularJS version was abusing `id` and using them when we knew it won't be unique.  In this case, the `id` value that you gave is not unique since there might be multiple range pickers of the same type on the page.

Reply:
removed ids